File: account/user-settings/update-user-profile-image/code-samples/updateUserProfileImage.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# PATH PARAMETERS
extensionId = '<ENTER VALUE>'
accountId = '<ENTER VALUE>'

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])

builder = rcsdk.create_multipart_builder()

image = ('alice_smith.gif', open('alice_smith.gif','r').read(), 'image/gif')
builder.add(image)

request = builder.request(f'/restapi/v1.0/account/{accountId}/extension/{extensionId}/profile-image')

resp = platform.send_request(request)

File: account/user-settings/update-user-profile-image/code-samples/updateUserProfileImage.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$extensionId = '<ENTER VALUE>';
$accountId = '<ENTER VALUE>';

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));

$request = $rcsdk->createMultipartBuilder()
    ->add('image', 'alice_smith.gif');
    ->add(fopen('./alice_smith.gif', 'r'))
    ->request("/restapi/v1.0/account/{$accountId}/extension/{$extensionId}/profile-image");

$r = $platform->sendRequest($request);
?>
